outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
```

Comment:
Although with all of the sync API stuff and possibly more details, it seems like this might not be the right place for this information. @tmeasday, do you have any thoughts on where information about NPM, Fibers, etc should go?